### ISSUE TITLE
Broken keyboard shortcut for text selection

### ISSUE THREAD

**Issue by dberry93** [WordPress/gutenberg#44563]
### Description

Deleting content by first using the keyboard shortcut (shift + arrow key) to select content spanning multiple blocks results in the incorrect amount of content being deleted and leaving the editor's DOM in an unstable state that can result in editor errors.

### Step-by-step reproduction instructions

Scenario 1:

1. Edit a post which contains multiple blocks
2. Place cursor at the start of the first block
3. Use keyboard shortcut to select content across multiple blocks (shift + down arrow)
4. Press backspace/delete

Scenario 2:

1. Place cursor at start of first block
2. Use the keyboard shortcut shift + alt + down arrow to select multiple blocks
3. Delete selected blocks
4. Move cursor to end of current block
5. Create new block by pressing enter resulting in an editor crash

### Screenshots, screen recording, code snippet

Scenario 1:

![wordpress-broken-text-selection](https://user-images.githubusercontent.com/57419717/192950352-d899af29-f2c9-4fce-9b8e-ef5145077100.gif)

Scenario 2:

![wordpress-broken-text-selection-editor-crash](https://user-images.githubusercontent.com/57419717/192960438-4bb0025f-46ce-48f3-8d74-131a88b89431.gif)

### Environment info

- Stock WordPress v6.0.2
- Docker image: `wordpress:latest`
- Mac OS X 12.6
- Google Chrome 105.0.5195.125

**docker-compose.yml:**

```
version: '3.1'

services:
  wordpress:
    image: wordpress
    restart: always
    ports:
      - 8080:80
    environment:
      WORDPRESS_DB_HOST: db
      WORDPRESS_DB_USER: admin
      WORDPRESS_DB_PASSWORD: admin
      WORDPRESS_DB_NAME: wordpress
    volumes:
      - wordpress:/var/www/html
  db:
    image: mysql:5.7
    restart: always
    environment:
      MYSQL_DATABASE: wordpress
      MYSQL_USER: admin
      MYSQL_PASSWORD: admin
      MYSQL_RANDOM_ROOT_PASSWORD: '1'
    volumes:
      - db:/var/lib/mysql

volumes:
  wordpress:
  db:
```

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by JeromeErasmus** [comment]
Yes confirmed. Please can we get this issues fixed

**Comment by ab49** [comment]
Yes confirmed. Please can we get this issues fixed

**Comment by BhupenT** [comment]
Yes i am also facing this issue. Would love this to get it resolved.

**Comment by talldan** [comment]
Thanks for reporting this. I think this is the same problem as #42213, so I'll close it as a duplicate.

The plan currently is to get a fix in place for WordPress 6.1, but a bug fix hasn't materialised yet. It is marked as high priority, so hopefully it will be fixed.
